extensions/emmet/src/abbreviationActions.ts

```diff
@@ -236,12 +247,80 @@ function doWrapping(_: boolean, args: any) {
 		return '';
 	}
 
-	const abbreviationPromise: Thenable<string | undefined> = (args && args['abbreviation']) ?
-		Promise.resolve(args['abbreviation']) :
-		vscode.window.showInputBox({ prompt: 'Enter Abbreviation', validateInput: inputChanged });
-	return abbreviationPromise.then(inputAbbreviation => {
-		return makeChanges(inputAbbreviation, true);
-	});
+	function getHistoryAsQuickPickItems(): vscode.QuickPickItem[] {
+		return Array.from(previousItems)
+			.map(item => {
+				const quickPickItem: vscode.QuickPickItem = {
+					label: item
+				};
+				return quickPickItem;
+			});
+	}
+
+	function getAbbreviationFromQuickPick(): Thenable<string> {
+		return new Promise<string>((resolve) => {
+			const pick = vscode.window.createQuickPick();
+			let valueToUse = '';
+			let accepted = false;
+			let changeRecommendations = true;
+			pick.title = localize('emmetEnterAbbreviationForWrap', 'Enter abbreviation, or choose a previous one');
+			pick.canSelectMany = false;
+			pick.items = getHistoryAsQuickPickItems();
+
+			pick.onDidChangeValue(e => {
+				valueToUse = e;
+				if (changeRecommendations) {
+					// create items to display in the quick pick
+					let items = getHistoryAsQuickPickItems();
+					if (valueToUse) {
+						// filter items like a search box
+						items = items.filter(item => item.label.startsWith(valueToUse));
+					}
+					pick.items = items;
+				}
+				changeRecommendations = true;
+				// update editor preview
+				inputChanged(valueToUse);
+			});
+			pick.onDidChangeActive((e) => {
+				// firing this event means the user has selected a value
+				// to fill the input box with via arrow keys
+				if (e.length) {
+					pick.value = e[0].label;
```

Comment:
actually, this event is also fired when the box first shows up, so there needs to be something to prevent that.